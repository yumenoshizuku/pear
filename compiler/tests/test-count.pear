Main()(
window.Window(),
window.Position("center"),
window.Size(250, 180),
window.Title("+-"),
frame.Fixed(),
window.Contain(frame),
plus.Button("+"),
plus.Size(80, 35),
plus.FixPos(frame, 50, 20),
minus.Button("-"),
minus.Size(80, 35),
minus.FixPos(frame, 50, 80),
label.Label("0"),
label.FixPos(frame, 190, 58),
window.Showall(),
plus.ClickIncrement(clickincr(label)),
minus.ClickDecrement(clickdecr(label)),
)
