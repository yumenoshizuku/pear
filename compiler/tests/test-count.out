#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>
#include <glib.h>
#include <glib/gstdio.h>


void clickdecr(GtkWidget* widget, gpointer label)
{
	char str[10];
	sprintf(str, "%d", atoi(gtk_label_get_text(GTK_LABEL(label))) - 1);
	gtk_label_set_text (GTK_LABEL(label), str);
}


void clickincr(GtkWidget* widget, gpointer label)
{
	char str[10];
	sprintf(str, "%d", atoi(gtk_label_get_text(GTK_LABEL(label))) + 1);
	gtk_label_set_text (GTK_LABEL(label), str);
}


int main(int argc, char** argv)
{
	GtkWidget* window;
	GtkWidget* frame;
	GtkWidget* plus;
	GtkWidget* minus;
	GtkWidget* label;
	
	gtk_init(&argc, &argv);
	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
	g_signal_connect(window, "destroy", G_CALLBACK(gtk_main_quit), NULL);
	gtk_window_set_position(GTK_WINDOW(window), GTK_WIN_POS_CENTER);
	gtk_window_set_default_size(GTK_WINDOW(window), 250, 180);
	gtk_window_set_title(GTK_WINDOW(window), "+-");
	frame = gtk_fixed_new();
	gtk_container_add(GTK_CONTAINER(window), frame);
	plus = gtk_button_new_with_label("+");
	gtk_widget_set_size_request(plus, 80, 35);
	gtk_fixed_put(GTK_FIXED(frame), plus, 50, 20);
	minus = gtk_button_new_with_label("-");
	gtk_widget_set_size_request(minus, 80, 35);
	gtk_fixed_put(GTK_FIXED(frame), minus, 50, 80);
	label = gtk_label_new ("0");
	gtk_fixed_put(GTK_FIXED(frame), label, 190, 58);
	gtk_widget_show_all(window);
	g_signal_connect(G_OBJECT(plus), "clicked", G_CALLBACK(clickincr), label);
	g_signal_connect(G_OBJECT(minus), "clicked", G_CALLBACK(clickdecr), label);
	gtk_main();
	return 0;
}

